sdk/tests/conformance2/buffers/buffer-data-and-buffer-sub-data-sub-source.html

```diff
@@ -0,0 +1,207 @@
+<!--
+
+/*
+** Copyright (c) 2016 The Khronos Group Inc.
+**
+** Permission is hereby granted, free of charge, to any person obtaining a
+** copy of this software and/or associated documentation files (the
+** "Materials"), to deal in the Materials without restriction, including
+** without limitation the rights to use, copy, modify, merge, publish,
+** distribute, sublicense, and/or sell copies of the Materials, and to
+** permit persons to whom the Materials are furnished to do so, subject to
+** the following conditions:
+**
+** The above copyright notice and this permission notice shall be included
+** in all copies or substantial portions of the Materials.
+**
+** THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<link rel="stylesheet" href="../../resources/js-test-style.css"/>
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"></script>
+</head>
+<body>
+<div id="description"></div>
+<div id="console"></div>
+
+<script>
+"use strict";
+description("Test bufferData/bufferSubData with ArrayBufferView sub source input");
+
+var testCases = [
+    {type: "Uint8Array",   size: 1},
+    {type: "Int8Array",    size: 1},
+    {type: "Int16Array",   size: 2},
+    {type: "Uint16Array",  size: 2},
+    {type: "Uint32Array",  size: 4},
+    {type: "Int32Array",   size: 4},
+    {type: "Float32Array", size: 4},
+    {type: "Float64Array", size: 8}
+];
+
+var wtu = WebGLTestUtils;
+var gl = wtu.create3DContext(undefined, undefined, 2);
+if (!gl) {
+    testFailed("WebGL context does not exist");
+} else {
+    testPassed("WebGL context exists");
+
+    bufferDataTest();
+    bufferSubDataTest();
+}
+
+function verifyBufferData(testCase, sourceByteOffset, size, data) {
+    var readbackBuffer = new ArrayBuffer(testCase.size * size);
+    var readbackView = new window[testCase.type](readbackBuffer);
+    gl.getBufferSubData(gl.ARRAY_BUFFER, sourceByteOffset, readbackView);
+    var pass = true;
+    var offset = sourceByteOffset / testCase.size;
+    for (var ii = 0; ii < size; ++ii) {
+        if (readbackView[ii] != data[ii]) {
+            testFailed("expected data at " + ii + ": " + data[iit] + ", got " + readbackView[ii]);
+            pass = false;
+        }
+    }
+    if (pass) {
+        testPassed("buffer data uploaded correctly");
+    }
+}
+
+function bufferDataTest() {
```

Comment:
I think it'd be cleaner if this took "testCases" as an argument instead of referencing it as a global variable. Would you make that change? Here and for bufferSubDataTest.